node/src/tests/test-PlainTransport.ts

```diff
@@ -316,6 +316,40 @@ test('router.createPlainTransport() with non bindable IP rejects with Error', as
 		.toThrow(Error);
 }, 2000);
 
+if (process.platform === 'linux')
+{
+	test('router.createPlainTransport() with enableMulticast succeeds', async () =>
+	{
+		let transport1: mediasoup.types.PlainTransport | undefined = undefined;
+		let transport2: mediasoup.types.PlainTransport | undefined = undefined;
+
+		await expect(async () =>
+		{
+			transport1 = await router.createPlainTransport(
+				{
+					listenIp        : '224.0.0.1',
+					enableMulticast : true
+				});
+
+			transport2 = await router.createPlainTransport(
+				{
+					listenIp        : '224.0.0.1',
+					enableMulticast : true
+				});
+		}).not.toThrow();
```

Comment:
What is this test asserting? transport1 and transport2 are using random listening UDP ports. This test passes in macOS in v3 branch without any multicast setup.

Reply:
This test should be as follows:

```ts
test('router.createPlainTransport() with enableMulticast succeeds', async () =>
{
	let transport1: mediasoup.types.PlainTransport | undefined;
	let transport2: mediasoup.types.PlainTransport | undefined;

	await expect(async () =>
	{
		const multicastIp = '224.0.0.1';
		const port = await pickPort({ ip: multicastIp, reserveTimeout: 0 });

		transport1 = await router.createPlainTransport(
			{
				listenInfo      :
				{
					protocol : 'udp',
					ip       : multicastIp,
					port     : port
				},
				enableMulticast : true
			});

		transport2 = await router.createPlainTransport(
			{
				listenInfo      :
				{
					protocol : 'udp',
					ip       : multicastIp,
					port     : port
				},
				enableMulticast : true
			});
	}).not.toThrow();

	transport1?.close();
	transport2?.close();
}, 2000);
```

And surprise, by adding the `UV_UDP_REUSEADDR`, this test **also** passes on macOS (because it's BSD which also supports multicast). So why is this PR hardcoding `if Linux` everywhere to enable multicast?

This should be done in `worker/meson.build` by defining a macro, the same way we pass `-DMS_LIBURING_SUPPORTED`.

Reply:
Indeed, we are playing `libuv` here which is multi-platform and does, as shown in the code, use this flag for Windows too. https://github.com/libuv/libuv/blob/master/src/win/udp.c#L559

Meaning this feature needs not to live under an activation flag.